Sources/PersistentCollections/PersistentDictionary/PersistentDictionary+MapValues.swift

```diff
@@ -22,17 +22,13 @@ extension PersistentDictionary {
   public func compactMapValues<T>(
     _ transform: (Value) throws -> T?
   ) rethrows -> PersistentDictionary<Key, T> {
+    // FIXME: We could do this as a structural transformation.
```

Comment:
👍

Reply:
Consider that structural formations must deal with compaction in order to guarantee a canonical representation, which slightly complicates the matter.

Reply:
It certainly does. Trying to do this in place (for `Set.formIntersection` and the like) would probably be a fool's errand, but I think an algorithm that constructs a new instance should be reasonably doable! (This is next on my list for part 9, actually. 🤓)